### PR TITLE
vendor: update github.com/coreos/go-oidc client package

### PR DESCRIPTION
**Pull request by ericchiang**
This change updates the github.com/coreos/go-oidc package to it's latest commit
(since we don't version that package).

Notable changes:
- Throw out JWTs with invalid claims early (coreos/go-oidc#97, brougt up in #30457)
- Remove the capnslog dependency (coreos/go-oidc#95)
- Support for Azure AD oddities (coreos/go-oidc#87)

cc @kubernetes/sig-auth

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31564)

### CODE REVIEW

**Comment by deads2k** [comment]
@ericchiang post-1.4 I presume?  Bumping dependencies seems fine, but I'd prefer to do it at the start of a cycle.

File: vendor/github.com/coreos/go-oidc/oauth2/oauth2.go [hunk 1]
```diff
@@ -332,16 +332,16 @@ func parseTokenResponse(resp *http.Response) (result TokenResponse, err error) {
...
-			Expires      int    `json:"expires"`
-			Error        string `json:"error"`
-			Desc         string `json:"error_description"`
+			AccessToken  string      `json:"access_token"`
+			TokenType    string      `json:"token_type"`
+			IDToken      string      `json:"id_token"`
+			RefreshToken string      `json:"refresh_token"`
+			Scope        string      `json:"scope"`
+			State        string      `json:"state"`
+			ExpiresIn    json.Number `json:"expires_in"` // Azure AD returns string
```
Comment (deads2k) [inline vendor/github.com/coreos/go-oidc/oauth2/oauth2.go]: Of course they would.  Surely something like https://tools.ietf.org/html/rfc6749#appendix-A.14 saying `1*DIGIT` is just a suggestion.

@liggitt does this affect us?

**Comment by deads2k** [comment]
lgtm post-1.4

**Comment by ericchiang** [comment]
yes, please mark as 1.5

**Comment by deads2k** [comment]
I marked for release note because of the active directory bit.

**Comment by ericchiang** [comment]
@k8s-bot test this #ISSUE

**Pull request merged**